RR.App.Tests/Controllers/HRIS/EmployeeSalaryDetailsControllerUnitTests.cs

```diff
@@ -0,0 +1,281 @@
+                               .ThrowsAsync(new Exception("An error occurred while deleting employee salary information."));
+
+        var result = await controller.DeleteSalary(employeeSalaryDetailsDto.Id);
+
+        var notFoundResult = Assert.IsType<NotFoundObjectResult>(result);
+        Assert.Equal("An error occurred while deleting employee salary information.", notFoundResult.Value);
+    }
+
+    [Fact]
+    public async Task UpdateEmployeeSalaryValidInputReturnsOkResult()
+    {
+        var employeeSalaryDetailsServiceMock = new Mock<IEmployeeSalarayDetailsService>();
+        var employeeServiceMock = new Mock<IEmployeeService>();
+        var controller = new EmployeeSalaryDetailsController(employeeSalaryDetailsServiceMock.Object);
+
+        var employeeSalaryDetailsDto = new EmployeeSalaryDetailsDto
+        {
+            Id = 1,
+            EmployeeId = 1,
+            Salary = 2000,
+            MinSalary = 1500,
+            MaxSalary = 3000,
+            Remuneration = 2500,
+            Band = EmployeeSalaryBand.Level1,
+            Contribution = null
+        };
+
+        employeeServiceMock.Setup(x => x.GetEmployeeById(employeeSalaryDetailsDto.EmployeeId))
+                           .ReturnsAsync(EmployeeTestData.EmployeeDto);
+
+        employeeSalaryDetailsServiceMock.Setup(x => x.UpdateEmployeeSalary(employeeSalaryDetailsDto))
+                                        .ReturnsAsync(employeeSalaryDetailsDto);
+
+        var result = await controller.UpdateSalary(employeeSalaryDetailsDto);
+
+        Assert.IsType<OkObjectResult>(result);
+    }
+
+    [Fact]
+    public async Task UpdateEmployeeSalaryExceptionThrownReturnsNotFoundWithMessage()
+    {
+        var employeeSalaryDetailsServiceMock = new Mock<IEmployeeSalarayDetailsService>();
+        var employeeServiceMock = new Mock<IEmployeeService>();
+        var controller = new EmployeeSalaryDetailsController(employeeSalaryDetailsServiceMock.Object);
+
+        var employeeSalaryDetailsDto = new EmployeeSalaryDetailsDto
+        {
+            Id = 1,
+            EmployeeId = 1,
+            Salary = 2000,
+            MinSalary = 1500,
+            MaxSalary = 3000,
+            Remuneration = 2500,
+            Band = EmployeeSalaryBand.Level1,
+            Contribution = null
+        };
+
+        employeeSalaryDetailsServiceMock.Setup(x => x.UpdateEmployeeSalary(It.IsAny<EmployeeSalaryDetailsDto>()))
+                               .ThrowsAsync(new Exception("An error occurred while updating employee salary information."));
+
+        var result = await controller.UpdateSalary(employeeSalaryDetailsDto);
+
+        var notFoundResult = Assert.IsType<NotFoundObjectResult>(result);
+        Assert.Equal("An error occurred while updating employee salary information.", notFoundResult.Value);
+    }
+
+    [Fact]
+    public async Task GetAllEmployeeSalariesByEmployeeReturnsOkResultWithList()
```

Comment:
The naming of your test is confusing, it makes it sound like you're testing a getAll() method. You're only fetching the salaries of a specific employee? Rename the test to something like GetSalariesByEmployeePass().

Reply:
You've got a point, and you've got a solid name recommendation. Next commit will fix this.